sklearn/utils/extmath.py

```diff
@@ -688,21 +688,38 @@ def _incremental_mean_and_var(X, last_mean=.0, last_variance=None,
     # old = stats until now
     # new = the current increment
     # updated = the aggregated stats
+    if not isinstance(last_sample_count, np.ndarray):
+        last_sample_count *= np.ones(X.shape[1], dtype=np.float)
     last_sum = last_mean * last_sample_count
-    new_sum = X.sum(axis=0)
-
-    new_sample_count = X.shape[0]
+    sum_func = np.nansum if ignore_nan else np.sum
+    new_sum = sum_func(X, axis=0)
+    if not isinstance(new_sum, np.ndarray):
+        new_sum *= np.ones(X.shape[1], dtype=np.float)
+
+    new_sample_count = np.count_nonzero(~np.isnan(X), axis=0)
```

Comment:
This is the problem line: numpy < 1.12 does not support `axis` in `count_nonzero`. Unfortunately, it does not trigger a TypeError either, and just silently counts the total number of nonzeros. However, `np.sum(~np.isnan(X), axis=0)` will give the same result, as will `len(X) - np.sum(np.isnan(X), axis=0)`.